theano/compile/mode.py

```diff
@@ -333,7 +353,7 @@ def requiring(self, *tags):
                                               self.provided_optimizer)
         return self.__class__(linker=link, optimizer=opt.requiring(*tags))
 
-    def clone(self, link_kwargs=None, **kwargs):
+    def clone(self, link_kwargs={}, optimizer=None, **kwargs):
```

Comment:
Don't use a mutable structure as a default value for an argument.  This is asking for trouble.